### PR TITLE
More VisualObj update funcs

### PR DESCRIPTION
**Pull request by joshlory**

### CODE REVIEW

File: src/main/visual_objs.c [hunk 1]
```diff
@@ -534,11 +713,113 @@ void func_800B2880(struct VisualObj* arg0)
...
+        arg0->unk30 = &D_8011BF40;
+        arg0->unk42 = 0x7802;
+        arg0->unk40 = 0;
+        arg0->base.unk16 = 3;
+        arg0->unk5C = 0;
+        arg0->base.state++;
+    }
+    temp_v1_2 = player->cur_anim;
+    var_s1 = -((((temp_v1_2 + 0x1B) & 0xFF) < 2U) ^ 1);
+    if (((temp_v1_2 + 0x23) & 0xFF) < 2U) {
```
Comment (sozud) [inline src/main/visual_objs.c:787]: These ought to be something like
```
    if(temp_v1_2 > -3 && temp_v1_2 < 0x21) {
```
But I'm slightly off and the numbers are swapped.
https://decomp.me/scratch/nKL14

**Review by sozud — APPROVED** [review]
Nice work!

File: src/main/visual_objs.c (hunk 1 above)
Comment (joshlory) [reply src/main/visual_objs.c:787]: I tried a `switch` statement but no luck. Best I was able to find:

```c
    anim = player->cur_anim;
    var_s1 = -((anim == 0xE5 || anim == 0xE6) ^ 1);
    if (anim == 0xDD || anim == 0xDE) {
        var_s1 = 1;
    }
    if (anim == 0xDF || anim == 0xE0) {
        var_s1 = 2;
    }
    if (anim == 0xE1 || anim == 0xE2) {
        var_s1 = 3;
    }
    if (anim == 0xE3 || anim == 0xE4) {
        var_s1 = 4;
    }
```

The `^ 1` is probably fake but I couldn't get a ternary to match

Comment (sozud) [reply src/main/visual_objs.c]: Nice, looks good!

**Pull request merged**